tests/backup/backup_kubevirt_test.go

```diff
@@ -3288,3 +3290,404 @@ var _ = Describe("{CustomBackupRestoreWithKubevirtAndNonKubevirtNS}", Label(Test
+			freezeRuleUid, err = Inst().Backup.GetRuleUid(BackupOrgID, ctx, freezeRuleName)
+			log.FailOnError(err, "Failed to get freeze rule uid for rule %s", freezeRuleName)
+			unfreezeRuleName = fmt.Sprintf("vm-unfreeze-rule-%s", RandomString(4))
+			err = CreateRuleForVMBackup(unfreezeRuleName, allVMs, Unfreeze, ctx)
+			log.FailOnError(err, "Failed to create unfreeze rule %s for VMs - %v", unfreezeRuleName, allVMNames)
+			unfreezeRuleUid, err = Inst().Backup.GetRuleUid(BackupOrgID, ctx, unfreezeRuleName)
+			log.FailOnError(err, "Failed to get unfreeze rule uid for rule %s", unfreezeRuleName)
+		})
+
+		Step("Take backups during cordoning and draining the nodes in which VM is running", func() {
+			log.InfoD("Take backups during cordoning and draining the nodes in which VM is running")
+			k8sCore := core.Instance()
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, appCtx := range scheduledAppContexts {
+				scheduledNamespace := appCtx.ScheduleOptions.Namespace
+				vms, err := GetAllVMsInNamespace(scheduledNamespace)
+				log.FailOnError(err, "Failed to get VMs deployed")
+				for _, vm := range vms {
+					createBackup := func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+						err = CreateVMBackupWithValidation(ctx, backupName, vms, SourceClusterName, backupLocationName, backupLocationUID, appContextsToBackup,
+							labelSelectors, BackupOrgID, sourceClusterUid, freezeRuleName, freezeRuleUid, unfreezeRuleName, unfreezeRuleUid, true)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of VM backup and Validation of backup [%s]", backupName))
+						mu.Lock()
+						backupNames = append(backupNames, backupName)
+						backupNamespaceMap[backupName] = namespace
+						defer mu.Unlock()
+					}
+					log.Infof("Taking backup of VMs before cordoning the node")
+					wg.Add(1)
+					backupName := fmt.Sprintf("%s-%s-%v", "pre-cordon-backup", scheduledNamespace, time.Now().Unix())
+					go createBackup(backupName, scheduledNamespace)
+					wg.Wait()
+
+					log.Infof("Taking backup of VMs during cordoning the node")
+					wg.Add(1)
+					backupName = fmt.Sprintf("%s-%s-%v", "cordon-backup", scheduledNamespace, time.Now().Unix())
+					go createBackup(backupName, scheduledNamespace)
+					//Get the node where the vm is scheduled.
+					preNodeName, err := GetNodeOfVM(vm)
+					log.FailOnError(err, fmt.Sprintf("Failed to get nodename for the VM %s", vm.Name))
+					defer func() {
+						err = k8sCore.UnCordonNode(preNodeName, 1*time.Minute, 5*time.Second)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying uncordoning the node %s", preNodeName))
+					}()
+					err = k8sCore.CordonNode(preNodeName, 1*time.Minute, 5*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying cordoning the node %s", preNodeName))
```

Comment:
Better to do log.FailOnError for all error checks

Reply:
Done